crate/operator/create.py

```diff
@@ -408,7 +473,9 @@ def get_statefulset_crate_env(
             name="CRATE_HEAP_SIZE",
             value=str(
                 int(
-                    bitmath.parse_string_unsafe(node_spec["resources"]["memory"]).bytes
+                    bitmath.parse_string_unsafe(
+                        get_cluster_resource_limits(node_spec, resource_type="memory")
```

Comment:
Should this have a fallback key too?

Reply:
No, for `memory` the keys are the same, it was only required for `cpu` vs `cpus` (the old `resources.cpus` and now `resources.limits.cpu`)